spec/views/hq/sponsors/_form_spec.rb

```diff
@@ -0,0 +1,149 @@
+require 'rails_helper'
+require 'cgi'
+
+RSpec.describe "hq/sponsors/_form.html.haml", type: :view do
+  let(:user) { build_stubbed :user}
+  let(:sponsor_full) do 
+    build_stubbed :sponsor_full, agent: user,
+                   request_fulfilled: [true, false].sample
+  end
+  let(:sponsor_new) { Sponsor.new }
+
+  before :each do
+    assign :sponsor, sponsor_full
+    assign :statuses, Status.all
+    assign :sponsor_types, SponsorType.all
+    assign :organizations, Organization.all
+    assign :branches, Branch.all
+    assign :cities, [sponsor_full.city].unshift(Sponsor::NEW_CITY_MENU_OPTION)
+  end
+
+  specify 'has a form' do
+    render
+
+    assert_select 'form'
+  end
+
+  describe 'has a "Cancel" button' do
+    specify 'using an existing Sponsor record' do
+      allow(sponsor_full).to receive(:id).and_return sponsor_full.id
+      render
+
+      assert_select 'a[href=?]', hq_sponsor_path(sponsor_full.id), text: 'Cancel'
+    end
+
+    specify 'using a new Sponsor record' do
+      allow(sponsor_full).to receive(:id).and_return nil
+      render
+
+      assert_select 'a[href=?]', hq_sponsors_path, text: 'Cancel'
+    end
+  end
+
+  describe 'has form values' do
+    specify 'using an existing Sponsor record' do
+      allow(User).to receive(:pluck).and_return([user.user_name, user.id])
+      render
+
+      #fextfields
+      ["name", "requested_orphan_count", "start_date", "new_city_name", "address", "email", 
+       "contact1", "contact2", "additional_info"].each do |field|
+        assert_select "input#sponsor_#{field}" do
+          if sponsor_full[field]
+            assert_select "[value=?]", CGI::escape_html(sponsor_full[field].to_s) 
+          else
+            assert_select "[value]", false
+          end
+        end
+      end
```

Comment:
I made this to remove the repetitive tests. 
I am not a fan of this way because it is hard to read but I gave it a try. I can change it back if requested.
What do you think?